### DISCUSSION TITLE
Run cypress commands outside tests.

### DISCUSSION THREAD

**Discussion by PranabJain** [cypress-io/cypress#20680] [Questions and Help]
I have a use case in which I have to run cypress commands outside of 'it' block. This is required to gather data(count of widgets on the current view in UI) according to which dynamic tests would be run on each widget then .
This feature will not be provided by cypress as mentioned at :https://github.com/cypress-io/cypress/issues/9111
Is there any hack / workaround for the same ?

**Comment by jjhelguero** [comment]
Is this something that cannot be done inside a `before()` or `beforeEach()`?

**Comment by PranabJain** [comment]
It could be done inside before() but I need to access the result of those cy commands to run a for loop over it block to create dynamic tests.

**Reply by jjhelguero** [reply]
If you are only "gathering data" to get the number of times to loop, then you can save it as an alias and call it later to use in with `_.times()` lodash function.

**Comment by PranabJain** [comment]
Alias wont' work because, 
To access the alias , I need to write something as cy.get('@user').....
But this cannot work outside it or before block due to inherent issue with cypress .

**Reply by jjhelguero** [reply]
I don't quite understand your set up. Could you go into more detail and maybe provide a possible code example of what you are wanting to do?
